acme/db/nosql/wire.go

```diff
@@ -0,0 +1,117 @@
+package nosql
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/acme"
+	"github.com/smallstep/nosql"
+)
+
+type dbDpopToken struct {
+	ID        string    `json:"id"`
+	Content   []byte    `json:"content"`
+	CreatedAt time.Time `json:"createdAt"`
+}
+
+// getDBDpopToken retrieves and unmarshals an DPoP type from the database.
+func (db *DB) getDBDpopToken(ctx context.Context, orderId string) (*dbDpopToken, error) {
+	b, err := db.db.Get(dpopTokenTable, []byte(orderId))
+	if nosql.IsErrNotFound(err) {
+		return nil, acme.NewError(acme.ErrorMalformedType, "dpop %s not found", orderId)
+	} else if err != nil {
+		return nil, errors.Wrapf(err, "error loading dpop %s", orderId)
+	}
+	d := new(dbDpopToken)
+	if err := json.Unmarshal(b, &d); err != nil {
```

Comment:
Here, you pass a pointer (`&d`) of a pointer (`d`). You should use plainly `d` instead. It may work anyway due to unintended recursive reflection in `json.Unmarshal()`, though.